server_development/topics/providers.adoc

```diff
@@ -308,9 +308,56 @@ public class EjbExampleUserStorageProviderFactory
 
 {project_name} has the ability to execute scripts during runtime in order to allow administrators to customize specific functionalities:
 
-* OpenID Connect Script Protocol Mapper
-* OpenID Connect Script Authenticator
+* Authenticator
 * JavaScript Policy
+* OpenID Connect Protocol Mapper
+
+==== Authenticator
+
+Authentication scripts must at least provide one of the following functions:
+`authenticate(..)` which is called from `Authenticator#authenticate(AuthenticationFlowContext)`
+`action(..)` which is called from `Authenticator#action(AuthenticationFlowContext)`
+
+Custom `Authenticator` should at least provide the `authenticate(..)` function.
+The following script `javax.script.Bindings` are available for convenient use within script code.
```

Comment:
You can use the `javax.script.Bindings`script within the code.

Reply:
Fixing